### ISSUE TITLE
No metrics from custom exporter are available within Prometheus within another namespace

### ISSUE THREAD

**Issue by renepardon** [prometheus-operator/prometheus-operator#4203]
**What did you do?**
I've installed the `kube-prometheus-stack` within it's default namespace `kube-prometheus-stack` hosted on DigitalOcean through 1-click-app-install without changing any default values.

I have a service within another namespace. Lets call it namespace J1:

```
apiVersion: v1
kind: Service
metadata:
  labels:
    app: app1
    service: app-stats-0
  name: app-stats-0
  namespace: J1
  annotations:
    alpha.monitoring.coreos.com/non-namespaced: "true"
spec:
  type: ClusterIP
  ports:
    - port: 9888
      name: app-exporter-0
      protocol: TCP
      targetPort: 9888
  selector:
    app: app1
    "statefulset.kubernetes.io/pod-name": app-0
```

Within `kube-prometheus-stack` namespace I created this `ServiceMonitor`:

```
apiVersion: monitoring.coreos.com/v1
kind: ServiceMonitor
metadata:
  labels:
    app: app1
  name: app1
  namespace: kube-prometheus-stack
spec:
  jobLabel: app1
  selector:
    matchExpressions:
      - { key: app, operator: In, values: [app1] }
  namespaceSelector:
    matchNames:
      - J1
  endpoints:
    - path: /metrics
      port: "9888"
      interval: 15s
---
apiVersion: monitoring.coreos.com/v1
kind: Prometheus
metadata:
  name: app1
  namespace: kube-prometheus-stack
spec:
  serviceAccountName: kube-prometheus-stack-prometheus
  serviceMonitorSelector:
    matchLabels:
      app: app1
  resources:
    requests:
      memory: 400Mi
  enableAdminAPI: false
```

**Did you expect to see some different?**

When executing a curl command within one of the pods within J1 namespace, I can access the metrics in proper format for Prometheus but within Prometheus the metrics never appear.

I executed: `kubectl port-forward svc/kube-prometheus-stack-prometheus 9090 -n kube-prometheus-stack` and searched within "rules", "targets" and "service discovery" but couldn't find related services/pods/metrics.

**Environment**

DigitalOcean k8s cluster
Node Versions: 1.21.2-do.2
Prometheus Operator installation: https://marketplace.digitalocean.com/apps/kubernetes-monitoring-stack?iframe=true

* Prometheus Operator version:

    `_quay.io/prometheus-operator/prometheus-operator:v0.47.0_`

* Kubernetes version information:

    `Client Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.3", GitCommit:"ca643a4d1f7bfe34773c74f79527be4afd95bf39", GitTreeState:"clean", BuildDate:"2021-07-15T21:04:39Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"21", GitVersion:"v1.21.2", GitCommit:"092fbfbf53427de67cac1e9fa54aaa09a28371d7", GitTreeState:"clean", BuildDate:"2021-06-16T12:53:14Z", GoVersion:"go1.16.5", Compiler:"gc", Platform:"linux/amd64"}`

* Kubernetes cluster kind:

    DigitalOcean way - create Droplets, LoadBalancer and everything works magically ;)

* Manifests:

```
see above
```

* Prometheus Operator Logs:

```
level=info ts=2021-08-07T13:04:04.611198374Z caller=operator.go:1163 component=prometheusoperator msg="sync prometheus" key=kube-prometheus-stack/app1
level=info ts=2021-08-07T13:04:05.086452922Z caller=operator.go:1163 component=prometheusoperator msg="sync prometheus" key=kube-prometheus-stack/app1
level=info ts=2021-08-07T13:04:08.058854073Z caller=operator.go:1163 component=prometheusoperator msg="sync prometheus" key=kube-prometheus-stack/app1
```

**Anything else we need to know?**:

**Comment by fpetkovski** [comment]
Can you try setting the `port` field in the ServiceMonitor endpoint to `app-exporter-0`?

**Comment by renepardon** [comment]
Unfortunately it doesn't make a difference @fpetkovski

**Comment by paulfantom** [comment]
It looks like your RBAC configuration does not allow prometheus access to the other namespace.

**Comment by ammmze** [comment]
Wanna try swapping the selector from using `matchExpressions` to using `matchLabels`? I just did a bit of trial and error trying to figure out why my flux scraping has stopped working a few weeks ago (probably after some prometheus update). But switching from matchExpressions to matchLabels is what got mine to show up again.

Here was my before:
```yaml
---
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: flux-system
  namespace: flux-system
  labels:
    app.kubernetes.io/part-of: flux
spec:
  namespaceSelector:
    matchNames:
      - flux-system
  selector:
    matchExpressions:
      - key: app
        operator: In
        values:
          - helm-controller
          - source-controller
          - kustomize-controller
          - notification-controller
          - image-automation-controller
          - image-reflector-controller
  podMetricsEndpoints:
    - port: http-prom
```

And after (just for helm-controller...though i'll need to setup more monitors for other apps)
```yaml
---
apiVersion: monitoring.coreos.com/v1
kind: PodMonitor
metadata:
  name: flux-system
  namespace: flux-system
  labels:
    app.kubernetes.io/part-of: flux
spec:
  namespaceSelector:
    matchNames:
      - flux-system
  selector:
    matchLabels:
      app: helm-controller
  podMetricsEndpoints:
    - port: http-prom
```

**Comment by slashpai** [comment]
@ammmze which version of Prometheus operator are you using? Thete was a bug in matchExpressions in 0.52 but that was fixed in 0.52.1

**Comment by ammmze** [comment]
> @ammmze which version of Prometheus operator are you using? Thete was a bug in matchExpressions in 0.52 but that was fixed in 0.52.1

Ah...thanks! I'll check next time I'm at my desk. I'm using kube-prometheus-stack helm chart at 21.0.0. Need to look to see what version of the operator that pulls in. I am using renovate and flux to keep things up to date though, so 🤷‍♂️
